doc/API-documentation.rst

```diff
@@ -0,0 +1,104 @@
+=================
+API Documentation
+=================
+
+:class:`GLM` Objects
```

Comment:
woops, there is a sphinx extension for this so you don't have to do the hard work ;) 

sorry I didn't realize you would jump into it! Here you go: http://www.sphinx-doc.org/en/stable/ext/autodoc.html